### DISCUSSION TITLE
Multiple ConfigurationKey for Consul Store KV

### DISCUSSION THREAD

**Discussion by Stronhold** [ThreeMammals/Ocelot#1813] [Ideas]
I would like to know if there can be multiple configuration keys to access different configuration files in KV Store or if I can match them using a pattern like it is done in [issue 296](https://github.com/ThreeMammals/Ocelot/issues/296)

**Comment by TomPallister** [comment]
Hi @Stronhold please see docs [here](http://ocelot.readthedocs.io/en/latest/features/configuration.html#configuration-key) this might help!

**Comment by Stronhold** [comment]
Hi,

I checked that before. As i Saw, there would be only one configurationkey at the same time or am I wrong?

Thanks in advance

**Comment by TomPallister** [comment]
@Stronhold you can one key per deployment of Ocelot. You cannot merge multiple configurations from Consul at the moment. If you would like to implement this I can support you! 😄 

edited: please ignore initial comment.

**Comment by Stronhold** [comment]
That was the point! Any help would be appreciated. 
Morover, i would like yo know if when we use the administration API, the config.json file, could be overwritten? As far as I have tested, this does not happen 

Thanks in advance!

**Comment by TomPallister** [comment]
@Stronhold are you using admin api with consul storing your configuration? If so the json file is not changed. If you are not using consul then it will be changed.

**Comment by Stronhold** [comment]
Nope. The idea would be using multiple configurationkey and use consul to have the routes. Through the administration API chance  the config.json. 

Any support to have multiple configurationkey is appreciated :)

**Comment by TomPallister** [comment]
Sorry @Stronhold I am very confused by what it is you want to implement. Please could you add as much detail as possible so I can understand.

**Comment by Stronhold** [comment]
I want to be able to have múltiple configurationkey to access to different consul configurations and merge them

**Comment by TomPallister** [comment]
this feature implement config merging for consul, same logic we use for file merging but applied to consul. put abstraction around logic so we can use same for both concrete implementations.

**Comment by marrop** [comment]
This might be slightly off topic but here goes. Is it possible via Ocelot to read a consul value from Consul's Key/Value store? Or is the only value one may read from Consul the configuration ("InternalConfiguration" or set by "GlobalConfiguration/ConfigurationKey"?)

**Comment by raman-m** [comment]
@ggnaegi  What is your opinion on this crazy idea with the Consul config? 😄

**Comment by ggnaegi** [comment]
> @ggnaegi  What is your opinion on this crazy idea with the Consul config? 😄 

I think it would be a nice addition, at least provide a custom key for the ocelot config stored in key/value store.

But then merge config... Like appsettings?

**Reply by raman-m** [reply]
So... We have new `AddOcelot` method which was not available in 2018 😄 and it takes `FileConfiguration` object, see below
- `public static IConfigurationBuilder AddOcelot(this IConfigurationBuilder builder, FileConfiguration fileConfiguration)`
- Link: [ConfigurationBuilderExtensions.AddOcelot](../blob/develop/src/Ocelot/DependencyInjection/ConfigurationBuilderExtensions.cs#L105)

The author (anyone) can save config files and read them in/from any persistent storage and apply this reading and merging in app startup logic...
Tom was right, the author wants too much! 🤣 He should provide clear user scenario first... If Consul will be persistent storage OK... Let's develop this. I hope, author will return to us with a PR, otherwise we will loose our time for the next rare user case. 😠

**Comment by raman-m** [comment]
@Stronhold  Dear Sergio!
Your request confuses all parties including Tom.
What is your user scenario?

Also, I see that [your repos](https://github.com/Stronhold?tab=repositories) have no Ocelot forked repo.
So, forked [Ocelot repo](https://github.com/Stronhold/Ocelot) doesn't exist in your account!
I would say you  violated Ocelot [development process](https://ocelot.readthedocs.io/en/latest/building/releaseprocess.html) which says: discuss -> issue -> PR.
But working on an issue means you have to fork Ocelot repo.
🆗 First fork, then request...

**Reply by raman-m** [reply]
@Stronhold Hi! And what's up, man?
